### PR TITLE
Enabling periodic builds

### PR DESCRIPTION
**Pull request by jvcoutinho**
This PR attempts to solve #59. It addresses these changes:

- Travis now executes a different test weekly on branch master (cron jobs).
- A new shell test was created for this purpose.

### CODE REVIEW

**Comment by jvcoutinho** [comment]
Notes:
- I've created a [new repository](https://github.com/jvcoutinho/s3m-test-samples), but I believe @pauloborba or @guilhermejccavalcanti should be its owner.
- The new shell test is inside `shelltests` directory, but another possibility is to put it along the new samples, once it doesn't belong to the test suite.
- A specific stat caught my attention: time. There are currently 1797 Java files. While unstructured merge took 119 seconds (2 minutes) to merge all of them, semistructured merge took 3050 seconds (51 minutes). It's a lot of time (and I had to reduce the sample size because Travis was accusing timeout).

File: testfiles/shelltests/s3mTravisCronJobsTests.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	assertTrue "[ '$FP_REDUCTION' = '100.00%' ] || [ '$FP_REDUCTION' = '100,00%' ]"
+
+	FN_REDUCTION=$(cat jfstmerge.summary | grep -Eo "And a reduction of [0-9]+[.,][0-9]+%" | grep -Eo "[0-9]+[.,][0-9]+%")
+	assertTrue "[ '$FN_REDUCTION' = '100.00%' ] || [ '$FN_REDUCTION' = '100,00%' ]"
+}
+
+testMergeRevisionsSample()
+{
+    # Configuration phase.
+    cp -r ${START_PATH}/sample/sampleleft/* .
```
Comment (jvcoutinho) [inline testfiles/shelltests/s3mTravisCronJobsTests.sh]: @guilhermejccavalcanti, aqui que eu copio os arquivos.
A proposta é pra já usar o git add nesses arquivos ao invés de copiar?

**Comment by gaabs** [comment]
@jvcoutinho , do you have a travis log output of those tests?

As for the execution time, I believe that this is not far from what @guilhermejccavalcanti observed. At https://dl.acm.org/citation.cfm?doid=3152284.3133883, he says that "
Taking the median of the measured times, the improved tool took approximately 24 minutes
to merge the entire sample, compared to only 45 seconds of the unstructured merge tool".

But it would be interesting to investigate where performance could be most improved.

File: testfiles/shelltests/s3mTravisCronJobsTests.sh (hunk 1 above)
Comment (guilhermejccavalcanti) [reply testfiles/shelltests/s3mTravisCronJobsTests.sh]: Isso. Ou então, talvez melhor, em vez de esses arquivos de testes estarem no nosso repositório, poderia haver um repositório de teste já com esses commits que você faz durante o teste. Assim, o merge ficaria trivial, sem necessidade de cópia dos arquivos, fazendo apenas checkouts e merge.  Só precisaria dos SHAs do left, base e right. É o que faço aqui https://github.com/guilhermejccavalcanti/GitMergesMiner/blob/travis/src/main/java/Extractor.groovy#L809

**Comment by pauloborba** [comment]
@guilhermejccavalcanti can we integrate that?

**Comment by guilhermejccavalcanti** [comment]
@jvcoutinho could you please test if this is still working properly, so I can integrate?

**Comment by jvcoutinho** [comment]
@pauloborba @guilhermejccavalcanti Actually the build is already enabled. It runs every wednesday. This code simply changes the weekly build from the regular to a special one.

**Comment by pauloborba** [comment]
@guilhermejccavalcanti please review and, if ok, integrate

**Pull request merged**